CppCoreGuidelines.md

```diff
@@ -5968,26 +5969,30 @@ You can handle self-assignment by explicitly testing for self-assignment, but of
 
     Foo& Foo::operator=(const Foo& a)   // OK, but there is a cost
     {
-        if (this == &a) return *this;
-        s = a.s;
-        i = a.i;
+        if (this != &a) {
```

Comment:
contradicts NR.2 "single-return rule makes it harder to concentrate error checking at the top of a function"

Reply:
I doubt that NR.2 was meant to imply "Every function _must_ have at least two return points." Using `if` for a block of code that you want to be executed conditionally is actually idiomatic C++.

Also, this is a "bad code" example anyway.

Reply:
I don't have an opinion on whether this is better or worse but I don't think "this is a bad code example" is a good justification. a "bad code" example should be bad in precisely one way so that it illustrates the point it is intended to illustrate without any distractions.

Reply:
It also contradicts F.56
"Shallow nesting of conditions makes the code easier to follow. It also makes the intent clearer. Strive to place the essential code at outermost scope, unless this obscures intent."

"Use a guard-clause to take care of exceptional cases and return early."